R/estimate_KM.R

```diff
@@ -65,102 +65,56 @@
 #' visR::estimate_KM(data = veteran_adam, strata = "trt")
 
 estimate_KM <- function(
-   data = NULL
-  ,strata = NULL
-  ,CNSR = "CNSR"
-  ,AVAL = "AVAL"
-  ,...
+    data = NULL
+    ,strata = NULL
+    ,CNSR = "CNSR"
+    ,AVAL = "AVAL"
+    ,...
 ){
-
-# Capture input to validate user input for data argument -----------------
+  # Capture input to validate user input for data argument ---------------------
   dots <- rlang::dots_list(...)
 
- ## Get actual data name as symbol
- ### Magrittre pipe returns "." which inactivates recalls to survfit in downstream functions
- ### map passes .x as Call$data
- ### df: catch expressions that represent base R subsets
-  Call <- as.list(match.call())
-  dfExpr <- Call[["data"]]
-
- ## Validate `data` and capture data name
-
-  if (is.null(data)) stop(paste0("Data can't be NULL."))
-
-  if (base::length(base::deparse(Call[["data"]])) == 1 && base::deparse(Call[["data"]]) %in% c(".", ".x")){
-    df <- the_lhs()
-    Call[["data"]] <- as.symbol(df)
-  } else {
-     df <- as.character(sub("\\[.*$", "", deparse(dfExpr))[1])
-  }
-
-  if (!(inherits(data, "data.frame") | inherits(data, "tibble") | inherits(data, "data.table"))) stop(paste0("Data can be of `class` dataframe or tibble or data.table."))
-
-  data <- as.data.frame(data)
-
-# Validate columns --------------------------------------------------------
```

Comment:
keep comments (throughout document)

Reply:
this still appears in the code, now on line 77